### PR TITLE
Feature/fundamentals 5

### PR DESCRIPTION
**Pull request by LeoKirasic**
- [x] You have read [The Odin Projects Contributing Guide](https://github.com/TheOdinProject/curriculum/blob/main/CONTRIBUTING.md).

#### 1.Describe the changes made:

I have added learning outcomes and knowledge checks with links to the fundamentals 5 lesson.

#### 2. If this PR is related to an open issue please reference it with the `#` operator and the issue number below:

#22466

### CODE REVIEW

File: foundations/javascript_basics/fundamentals-5.md [hunk 1]
```diff
@@ -28,3 +35,13 @@ At this point you just need a little more practice!  Go back to the [JavaScript
...
+
+### Knowledge Check
+
+This section contains questions for you to check your understanding of this lesson. If you’re having trouble answering the questions below on your own, review the material above to find the answer.
+
+- <a class="knowledge-check-link" href="https://medium.com/@zac_heisey/objects-vs-arrays-42601ff79421">What is the difference between objects and arrays?</a>
+- <a class="knowledge-check-link" href="https://developer.mozilla.org/en-US/docs/Learn/JavaScript/Objects/Basics#bracket_notation">How do you access object properties?</a>
+- <a class="knowledge-check-link" href="https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/map">What is _Array.prototype.map()_ useful for?</a>
+- <a class="knowledge-check-link" href="https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/Reduce">What is _Array.prototype.reduce()_ useful for?</a>
+- <a class="knowledge-check-link" href="https://www.c-sharpcorner.com/blogs/find-vs-filter-in-javascrtipt">What is the difference between _Array.prototype.find()_ and _Array.prototype.filter()_?</a>
```
Comment (wise-king-sullyman) [inline foundations/javascript_basics/fundamentals-5.md:47]: These linked resources are not otherwise linked in this lesson.

**Review by wise-king-sullyman — CHANGES_REQUESTED** [review]
I don't think we really want to have knowledge checks that link to resources not otherwise covered in the lesson, but I will get up with the team and discuss.

File: foundations/javascript_basics/fundamentals-5.md [hunk 2]
```diff
@@ -28,3 +35,12 @@ At this point you just need a little more practice!  Go back to the [JavaScript
...
 </div>
+
+### Knowledge Check
+
+This section contains questions for you to check your understanding of this lesson. If you’re having trouble answering the questions below on your own, review the material above to find the answer.
+
+- <a class="knowledge-check-link" href="https://javascript.info/object#summary">What is the difference between objects and arrays?</a>
+- <a class="knowledge-check-link" href="https://developer.mozilla.org/en-US/docs/Learn/JavaScript/Objects/Basics#bracket_notation">How do you access object properties?</a>
+- <a class="knowledge-check-link" href="https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/map">What is _Array.prototype.map()_ useful for?</a>
+- <a class="knowledge-check-link" href="https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/Reduce">What is _Array.prototype.reduce()_ useful for?</a>
```
Comment (wise-king-sullyman) [inline foundations/javascript_basics/fundamentals-5.md:46]: While we do link to MDN in general here we don't link to the array reference in this lesson. I'll have to check with the team but I don't think we want to have knowledge checks that aren't *explicitly* linked to if that makes sense.

**Review by wise-king-sullyman — CHANGES_REQUESTED** [review]

File: foundations/javascript_basics/fundamentals-5.md (hunk 2 above)
Comment (LeoKirasic) [reply foundations/javascript_basics/fundamentals-5.md:46]: I can change the link to the video where it was covered. I just thought that a different explanation from the docs would be beneficial.

Comment (wise-king-sullyman) [reply foundations/javascript_basics/fundamentals-5.md:46]: @LeoKirasic yeah we don't want to link to resources that we don't specifically tell learners to review as part of the lesson.

If we're going to link to an external page in the knowledge checks we need to link to that exact same page in the lesson content.

**Review by wise-king-sullyman — APPROVED** [review]
Thanks Leo!

**Pull request merged**